### PR TITLE
Enrollments export

### PR DESCRIPTION
**Pull request by islemaster**
[Update 5/18] Note that this PR completes the second "next step", introducing `bin/cron/enrollments_to_gdrive`. Will do the update of `bin/cron/summer_workshops_to_gdrive` in a separate PR.

[PLC-883](https://codedotorg.atlassian.net/browse/PLC-883): Work in progress.  Notes for @bencodeorg and @Hamms:

Please see the jira task for requirements, including the range of workshop enrollments to be exported and the expected column list / column order.

Add the following to your `[cdo]/config/development.yml.erb` to pull the staging service account credentials from Secrets Manager:

```
gdrive_export_secret: !Secret
applications_2020_2021_gsheet_key: !Secret
```

So far, this PR:

- Moved `lib/cdo/google_drive` to `lib/cdo/google/drive`
- Created `lib/cdo/google/sheet` which encapsulates the GSheet export and ACL check logic.
- Updated `bin/cron/teacher_applications_to_gdrive` to use `Google::Sheet`

Next steps probably include:

- Update `bin/cron/summer_workshops_to_gdrive` to use `Google::Sheet` (intentionally skips the ACL check)
- Introduce `bin/cron/enrollments_to_gdrive` (very rough draft included in the "WIP" commit here) which should export appropriate data and do an ACL check (since it includes PII)
  - You may need to introduce new config values for the new gsheet key and ACL exceptions.

### CODE REVIEW

File: bin/cron/enrollments_to_gdrive [hunk 1]
```diff
@@ -3,57 +3,65 @@ require_relative '../../lib/cdo/only_one'
...
+    sheet_name: 'enrollments',
+    rows: results
+  )
+end
 
-# The first row is an array of strings, one for each attribute name.
-workshops << Api::V1::Pd::WorkshopDownloadSerializer.new(Pd::Workshop.first).attributes.keys.map(&:to_s)
+def notify_of_external_sharing(sheet)
+  if sheet.external_emails_with_access.present?
+    email_domains = external_emails.map {|email| email.slice(/@.*/)}.uniq
```
Comment (Hamms) [inline bin/cron/enrollments_to_gdrive]: should this be

```ruby
external_emails = sheet.external_emails_with_access
if external_emails.present?
  email_domains = external_emails.map {|email| email.slice(/@.*/)}.uniq
```

File: bin/cron/summer_workshop_enrollments_to_gdrive [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  sheet.export(
+    sheet_name: 'enrollments',
+    rows: results
+  )
+end
+
+def notify_of_external_sharing(sheet)
+  if sheet.external_emails_with_access.present?
+    email_domains = external_emails.map {|email| email.slice(/@.*/)}.uniq
```
Comment (Hamms) [inline bin/cron/summer_workshop_enrollments_to_gdrive:52]: it might be nice as a follow-up to also move this "email domains" logic into somewhere reusable

**Review by Hamms — APPROVED** [review]
one nit, otherwise looks great!

File: bin/cron/summer_workshop_enrollments_to_gdrive (hunk 2 above)
Comment (bencodeorg) [reply bin/cron/summer_workshop_enrollments_to_gdrive:52]: Totally, I'll follow up to get the other script to use the `Google::Sheet` class and look at including this.

**Pull request merged**